### PR TITLE
Improved docs with custom visualization example

### PR DESCRIPTION
**Pull request by notdanilo**
### Pull Request Description
This PR adds an example for creating a custom JavaScript visualization.

### Important Notes
Four shortcuts were updated:
* Visualization cycling is now <kbd>Ctrl</kbd>+<kbd>f</kbd>
* Sending mock data is now <kbd>Ctrl</kbd>+<kbd>d</kbd>,
* Adding a new empty node is now <kbd>Ctrl</kbd>+<kbd>n</kbd>
* Removing selected nodes is now <kbd>Ctrl</kbd>+<kbd>Backspace</kbd>

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/luna/enso/blob/main/docs/style-guide/rust.md).
- [x] All code has been tested where possible.
- [x] All code has been profiled where possible.

### CODE REVIEW

**Comment by wdanilo** [comment]
The docs lack:
- detailed description what easch function in JS does
- that you can skip some functions
- some skipped API, like preprocessing function definition

I think the above are covered in the code docs

File: docs/product/shortcuts.md [hunk 1]
```diff
@@ -29,6 +29,16 @@
...
+
+
+
+### Visualization
+
+| Shortcut | Action |
+| -------- | ------ |
+| <kbd>cmd</kbd> / <kbd>ctrl</kbd> + <kbd>space<kbd> | Toggle visualization visibility of the selected node. |
+| <kbd>cmd</kbd> / <kbd>ctrl</kbd> + <kbd>f<kbd>     | Cycle visualizations of the selected node. |
+| <kbd>cmd</kbd> / <kbd>ctrl</kbd> + <kbd>d<kbd>     | Send test data to the selected node. |
```
Comment (mwu-tow) [inline docs/product/shortcuts.md:44]: I'd consider keeping the "debug only" shortcuts separate from the other ones.

File: docs/product/visualizations.md [hunk 2]
```diff
@@ -111,8 +111,58 @@ an Enso type to a set of visualizations defined for that type. The type might be
 very generic, like `[a]` (which in Enso terms means list of any elements).
 
 ### Defining a Visualization
-This needs to be described in detail. For now, we can just assume that the user
-is allowed to create visualization and register it in Enso.
+Visualizations is planned to be defined both with Enso and JavaScript but, for now, only JavaScript visualizations are
```
Comment (mwu-tow) [inline docs/product/visualizations.md:114]: Either this or the paragraph above is wrapped to a wrong column count limit.

File: docs/product/visualizations.md [hunk 3]
```diff
@@ -111,8 +111,58 @@ an Enso type to a set of visualizations defined for that type. The type might be
...
+
+    constructor(handle) {
+        super(handle);
+        this.canvas = document.createElement("canvas");
+        this.ctx    = this.canvas.getContext("2d");
+        this.dom.appendChild(this.canvas);
+    }
+    
+    onDataReceived(data) {
+        if (typeof data == "number") {
```
Comment (mwu-tow) [inline docs/product/visualizations.md:147]: What is the `data`? What user can expect this to be?

File: docs/product/visualizations.md [hunk 4]
```diff
@@ -11,141 +11,223 @@ tags: [product]
...
 - **Display results of nodes**  
-  Each node can be assigned with one or more visualization. After a node
-  computes its new value, the visualization shows it in an understandable way to
-  the user. Please note that a single node can be assigned with multiple
-  visualizations at the same time. For example, a node might want to display a
-  map of locations and their list at the same time next to each other.
+  Each node can be assigned with one or more visualization. After a node computes its new value, the visualization shows
+  it in an understandable way to the user. Please note that a single node can be assigned with multiple visualizations
+  at the same time. For example, a node might want to display a map of locations, and their list at the same time next
+  to each other.
```
Comment (wdanilo) [inline docs/product/visualizations.md:17]: This text should be reformatted back. We keep rust sourcecodewith 100 charslimit but everything else, including all docs, with 80 chars limit.

File: docs/product/visualizations.md [hunk 5]
```diff
@@ -11,111 +11,85 @@ tags: [product]
...
 - **Display results of nodes**  
-  Each node can be assigned with one or more visualization. After a node
-  computes its new value, the visualization shows it in an understandable way to
-  the user. Please note that a single node can be assigned with multiple
-  visualizations at the same time. For example, a node might want to display a
-  map of locations, and their list at the same time next to each other.
+  Each node can be assigned with one or more visualization. After a node computes its new value, the visualization shows
+  it in an understandable way to the user. Please note that a single node can be assigned with multiple visualizations
+  at the same time. For example, a node might want to display a map of locations, and their list at the same time next
+  to each other.
```
Comment (farmaazon) [inline docs/product/visualizations.md:17]: This and below paragraphs were extended to 120 chars per line. Please revert it back to 100

**Pull request merged**